### PR TITLE
Add a depend_files: keyword to windows.compile_resources() 

### LINKED ISSUE #2789: Add something like depend_files to windows.compile_resources()
Resource script can include various other files (bitmap, cursor, font, html, icon, message table, binary data, manifest), it would be nice if it were possible to declare the resource script depends on these.

### PR DESCRIPTION
**Pull request by jon-turney**
This is the change suggested in #2815, with tests and documentation added.

Fixes #2789 (duplicate #2830)

### CODE REVIEW

**Comment by jon-turney** [comment]
This fails with the MSVC backend.  I have no idea why. 

Opening the generated .sln in Visual Studio 2015 won't even show me the properties of that build step, which doesn't build confidence, but at first glance, it looks like `<Inputs>` in a `<CustomBuildStep>` is just being ignored.

**Comment by jon-turney** [comment]
> This fails with the MSVC backend. I have no idea why.

I fixed a couple of mistakes and now this seems to work in CI.  It still fails when I try to test it locally, but that could be something I'm doing wrong...

File: run_unittests.py [hunk 1]
```diff
@@ -735,9 +735,7 @@ def assertRebuiltTarget(self, target):
             self.assertIn('Linking target {}'.format(target), ret)
         elif self.backend is Backend.vs:
             # Ensure that this target was rebuilt
-            clre = re.compile('ClCompile:\n [^\n]*cl[^\n]*' + target, flags=re.IGNORECASE)
             linkre = re.compile('Link:\n [^\n]*link[^\n]*' + target, flags=re.IGNORECASE)
-            self.assertRegex(ret, clre)
```
Comment (nirbheek) [inline run_unittests.py]: Why was this removed?

Comment (jon-turney) [reply run_unittests.py]: Because it's wrong.  See the commentary on commit d29cba1

**Comment by jpakkane** [comment]
#3486 is merged, this one is ok to merge as well.
